pkgs/build-support/deno/fetch-deno-deps/default.nix

```
@@ -0,0 +1,226 @@
+{
+  callPackage,
+  stdenvNoCC,
+  lib,
+  cacert,
+}:
+let
+  inherit (callPackage ./scripts/deno/default.nix { }) fetch-deno-deps-scripts;
+  inherit (callPackage ./scripts/rust/file-structure-transformer-vendor/default.nix { })
+    file-structure-transformer-vendor
+    ;
+
+  transformDenoLock =
+    {
+      denoLock,
+    }:
+    lib.extendMkDerivation {
+      constructDrv = stdenvNoCC.mkDerivation;
+      extendDrvArgs =
+        _: args:
+        let
+          pname = args.pname or "";
+          version = args.version or "";
+        in
+        {
+          name = "denoDeps-transformedDenoLock-${pname}-${version}";
+          src = null;
+          unpackPhase = "true";
```

1. The arguments to the build helper should be in the function passed to `extendDrvArgs`. You can see an example [here](https://github.com/RossSmyth/press/blob/main/src/buildTypstDocument.nix)
2. Since this is a new interface, I would require `pname`, and `version`.
3. Rather than setting `unpackPhase` to `true`, just disable it.
4. Access other attribute via `finalAttrs` so that the fixed-point overrides propogate properly
```suggestion
    lib.extendMkDerivation {
      constructDrv = stdenvNoCC.mkDerivation;
      extendDrvArgs =
        finalAttrs: {
          pname ? "",
          version ? "",
          ...
        }@args:
        {
          name = "denoDeps-transformedDenoLock-${finalAttrs.pname}-${finalAttrs.version}";
          src = null;
          dontUnpack = true;
```